schema/mysql/schema.sql

```diff
@@ -0,0 +1,413 @@
+CREATE TABLE available_channel_type (
+    type varchar(255) NOT NULL,
+    name mediumtext NOT NULL,
+    version mediumtext NOT NULL,
+    author mediumtext NOT NULL,
+    config_attrs mediumtext NOT NULL,
```

Comment:
I've raised the text columns 64k -> 16M.

Reply:
That change doesn't really make sense for most of the columns. The `name` fields are names that users enter in Web, they won't type 64KB+ there. Author and version of a plugin will realistically stay much shorter.

The only columns where I can see that this change makes sense:

- `available_channel_type.config_attrs`: Ff a channel provides many config options and offers translations in all possible languages.
- `channel.config`: With many options set to long values, think of something like pasting multiple X.509 certificates into the config.
- `rotation.options`: maybe, for details you have to ask Web if there's a limit on how big this can get.
- `object.mute_reason`, `event.mute_reason`, `event.message`, `incident_history.message`: Those sound like ones where you want to have a reasonable limit as well. At some point, it's more likely to contain just garbage rather than useful information anyways. For scale, the GPLv2 is roughly 18KB in size, so I think that 64KB should already be more than enough, but having it as `mediumtext` would also allow limiting this to a different size at a later point without having to change the database schema.

Reply:
Definitely a problem of the Postgres schema, if any. But it isn't. If it was an actual problem, you<sub>(plural)</sub> had used varchar(x) in favour of text. The latter [can hold 1G](https://www.postgresql.org/docs/current/datatype-character.html), so my 16M here are peanuts.

```
idb=# update lolcat set s=s||s;
UPDATE 1
idb=# select length(s) from lolcat;
  length
-----------
 536870912
(1 row)

idb=# update lolcat set s=s||s;
ERROR:  invalid memory alloc request size 1073741828
idb=# \d lolcat
             Table "public.lolcat"
 Column | Type | Collation | Nullable | Default
--------+------+-----------+----------+---------
 s      | text |           |          |

idb=#
```

Reply:
The PostgreSQL schema uses `text` because it's the most obvious type given that we didn't bother spending time choosing the specific `x` for `varchar(x)`. However, I think it's obvious that for most columns, a reasonable choice for `x` will be way less than 65536. Thus, I really don't see a reason why `mediumtext` should be the standard choice for the MySQL/MariaDB schema.